packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -32,11 +32,15 @@ function QRCodeDialog() {
 
     useEffect(() => {
         if (directionsFrom && directionsTo) {
-            // The target URL when the user opens the QR code dialog
-            // Create the URL by combining the origin and the pathname
-            // Check if the pathname is a '/' before appending it.
-            let targetUrl = window.location.origin + (window.location.pathname === '/' ? '' : window.location.pathname)
+            // Get the last character of the pathname
+            const lastCharacter = window.location.pathname.slice(-1);
 
+            // Check if the pathname is a '/' before appending it to the targetUrl
+            let pathname = (lastCharacter === '/') ? window.location.pathname.slice(0, -1) : window.location.pathname;
```

Comment:
Going with a `replace` using a regular expression may be more elegant: https://stackoverflow.com/a/7602314

Either way, this needs some good testing.

Reply:
That's a good idea, I used the `replace` instead.